testing/trino-testing/src/main/java/io/trino/testing/BaseFailureRecoveryTest.java

```diff
@@ -126,13 +124,9 @@ protected final QueryRunner createQueryRunner()
                 ImmutableMap.of(
                         // making http timeouts shorter so tests which simulate communication timeouts finish in reasonable amount of time
                         "scheduler.http-client.idle-timeout", REQUEST_TIMEOUT.toString()),
-                new AbstractConfigurationAwareModule() {
```

Comment:
hmm.. that's why i wanted to test https://github.com/airlift/airlift/pull/1679 with a snapshot build.
i am worried there might be many places like this, so this gonna be painful.
maybe instead we can add some additional synchro or cloning. Or perhaps simply move the `AbstractConfigurationAwareModule` possession of the configuration factory to a thread local.

cc @electrum wdyt